Docs/Configuration.tex

```diff
@@ -2573,6 +2573,32 @@ \subsection{Quirks Properties}\label{kernelpropsquirks}
   on all systems but those explicitly dedicated to scientific or media calculations.
   Only certain Xeon models typically benefit from the patch.
 
+\item
+  \texttt{CustomPciSerialDevice}\\
+  \textbf{Type}: \texttt{plist\ boolean}\\
+  \textbf{Failsafe}: \texttt{false}\\
+  \textbf{Requirement}: 10.7\\
+  \textbf{Description}: Performs change of PMIO register base address
+  on a customised PCI serial device.
+
+  The patch changes the PMIO register base address from the default range \texttt{0x3F8-0x3FF}
+  to an actucal one retrieved by \texttt{GetSerialRegisterBase} function call, such that
+  the real PMIO address of the serial device will also be recognised by XNU.
+
+  \emph{Note}: In addition to this patch, kext \texttt{Apple16X50PCI0} should be prevented from attaching
+  to have \texttt{kprintf} method working properly. This can be achieved by setting (i.e. \texttt{Delete}, then \texttt{Add})
+  the \texttt{class-code} property of the PCI serial port device to \texttt{FFFFFFFF} in \texttt{DeviceProperties} section.
+  As an alternative solution, a codeless kext \texttt{PCIeSerialDisable.kext} shown in the spoiler \texttt{PCIeSerialDisable.kext/Contents/Info.plist} at
+  \href{https://github.com/acidanthera/bugtracker/issues/1954}{acidanthera/bugtracker\#1954}, may also be used.
+
+  \emph{Note 2}: For this patch to be correctly applied, both \texttt{Init} and \texttt{Override}
+  must be enabled with all keys properly set in \texttt{Custom}, under section \texttt{Misc->Serial}.
+
+  \emph{Note 3}: Since this patch does not provide MMIO support, \texttt{UseMmio} under section \texttt{Misc->Serial->Custom} must be disabled.
+
+  \emph{Note 4}: The serial baud rate must be correctly set in both \texttt{BaudRate} under section \texttt{Misc->Serial->Custom} and via
+  \texttt{serialbaud=VALUE} boot argument, both of which should match against each other.
```

Comment:
I will take a look whether we can set this boot argument automatically.

Reply:
Please do not. OpenCore should not interfere with such stuff.